lib/src/piece/assign.dart

```diff
@@ -19,37 +19,75 @@ import 'piece.dart';
 ///
 ///     var x = 123;
 ///
-/// If the value is a delimited "block-like" expression, then we allow splitting
-/// inside the value but not at the `=` with no additional indentation:
+/// This state also allows splitting the right side if it can be block
+/// formatted:
 ///
 ///     var list = [
 ///       element,
 ///     ];
 ///
-/// [_atOperator] Split after the `=`:
+/// [_blockSplit] Block split in the operands that support it and expression
+/// split in the others. This state requires at least one operand to support
+/// block splitting. Examples:
 ///
-///     var name =
-///         longValueExpression;
+///     var [
+///       element,
+///     ] = value;
+///
+///     var [
+///       element,
+///     ] = longOperand +
+///         anotherOperand;
+///
+///     var (longVariable &&
+///         anotherVariable) = [
+///       element,
+///     ];
+///
+/// [_atOperator] Split at the `=` or `in` operator and allow expression
+/// splitting in either operand:
+///
+///     var (longVariable &&
+///             anotherVariable) =
+///         longOperand +
+///             anotherOperand;
 class AssignPiece extends Piece {
-  /// Split after the operator.
+  /// Split at the operator.
```

Comment:
I liked the previous "after the operator". I think it's clearer to me where it actually splits.

Reply:
It doesn't always split after the operator, though. :) When an AssignPiece splits at the `in` keyword in a for-in loop, it splits before the `in`, not after.